eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Ratify.hs

```diff
@@ -135,6 +140,51 @@ accepted RatifyEnv {reStakePoolDistr = PoolDistr poolDistr} gas =
       HardForkInitiation {} -> 101 % 100
       _ -> spoThreshold
 
+dRepAccepted :: forall era. RatifyEnv era -> GovActionState era -> Bool
```

Comment:
I think it'd be better if we could reuse some of the logic across all the `*Accepted` functions. Right now it looks like there's some repetition between `dRepAccepted` and `spoAccepted`

Reply:
So for example the difference between `dRepAccepted` and `spoAccepted` is the map they use for looking up votes and what the default vote is when somebody hasn't submitted a vote, so we could make a function that takes both the map and the default vote and then use that function for both DReps and SPOs.

Reply:
The conditions are a bit different though, because for `spoAccepted` we already have the ratios, but here we have to compute them.  So I'm iterating the distribution and computing the ratios incrementally, whereas for spos we are iterating the votes, and then looking the corresponding stake up in the distribution. 
If we went with the same formula, we'd have to first add all the stake and then compute what needs to be removed and remove it.  But maybe there is a better way indeed